src/Microsoft.AspNet.Cors.Core/ICorsPolicyProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Cors.Core
+{
+    /// <summary>
+    /// A type which can provide a <see cref="CorsPolicy"/> for a particular <see cref="HttpContext"/>.
+    /// </summary>
+    public interface ICorsPolicyProvider
+    {
+        /// <summary>
+        /// Gets a <see cref="CorsPolicy"/> from the given <paramref name="context"/>
+        /// </summary>
+        /// <param name="context">The <see cref="HttpContext"/> associated with this call.</param>
+        /// <returns>A <see cref="CorsPolicy"/></returns>
+        CorsPolicy GetPolicy(HttpContext context);
```

Comment:
While in Web API we came across a scenario where users wanted to get the policy from database (ex: one could dynamically add/update trusted origin list in a database without changing code)...for this reason I *think* we had this call to be async...